oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/introspection/OAuth2TokenIntrospectionClient.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.introspection;
+
+import java.util.Map;
+
+/**
+ * Implementations of this interface are responsible for &quot;introspecting&quot;
+ * the contents of an OAuth 2.0 Token.
+ *
+ * A typical implementation would contact an
+ * <a href="https://tools.ietf.org/html/rfc7662" target="_blank">OAuth 2.0 Introspection Endpoint</a>
+ * hosted on an OAuth 2.0 Authorization Server.
+ *
+ * @author Josh Cummings
+ * @since 5.2
+ */
+public interface OAuth2TokenIntrospectionClient {
```

Comment:
I like the name change, but can you update the commit comment and PR description to reflect the name change?